### PR TITLE
FS-3535 : sqs extended client to handle larger messages while sending through SQS service

### PR DESCRIPTION
**Pull request by nuwan-samarasinghe**
[
_Add ticket reference to Pull Request title: e.g. 'FS-123: Add content', if there is no ticket prefix with BAU_](https://dluhcdigital.atlassian.net/browse/FS-3535)

### Change description
sqs extended client to handle larger messages while sending through SQS service

- [x] Unit tests and other appropriate tests added or updated
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### CODE REVIEW

File: fsd_utils/services/aws_extended_client.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+
+    def get_queues(self, prefix=None):
+        """
+        Gets a list of SQS queues. When a prefix is specified, only queues with names
+        that start with the prefix are returned.
+
+        :param prefix: The prefix used to restrict the list of returned queues.
+        :return: A list of Queue names.
+        """
+        if prefix:
```
Comment (NarenderRajuB) [inline fsd_utils/services/aws_extended_client.py:75]: can we move these if else to a small functions. would be better if use just use "if" easy to read and good practice

def get_queue_urls(self, prefix=None):
    if prefix:
       return self.sqs_client.list_queues(QueueNamePrefix=prefix)["QueueUrls"]
   return self.sqs_client.list_queues()["QueueUrls"]

queues = get_queue_urls(self, prefix=None)

File: fsd_utils/services/aws_extended_client.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+            MessageGroupId=message_group_id,
+            MessageDeduplicationId=message_deduplication_id,
+        )
+        # Check if the delete operation succeeded, if not raise an error?
+        status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+        if status_code != 200:
+            raise SQSExtendedClientException(
+                ExceptionMessages.FAILED_SUBMIT_MESSAGE.format(status_code)
+            )
+        else:
```
Comment (NarenderRajuB) [inline fsd_utils/services/aws_extended_client.py:122]: we can remove else here

File: fsd_utils/services/aws_extended_client.py [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+        if self.large_payload_support and self.always_through_s3:
+            extended_message = []
+            for msg in messages:
+                dict_msg = {
+                    "sqs": msg,
+                    "s3": self._retrieve_message_from_s3(msg["Body"]),
+                }
+                extended_message.append(dict_msg)
+            return extended_message
+        else:
```
Comment (NarenderRajuB) [inline fsd_utils/services/aws_extended_client.py:176]: same we can remove else here

File: fsd_utils/services/aws_extended_client.py [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+            response = self.sqs_client.delete_message_batch(
+                QueueUrl=queue_url, Entries=entries
+            )
+            # Check if the delete operation succeeded, if not raise an error?
+            status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+            if status_code != 200:
+                raise SQSExtendedClientException(
+                    ExceptionMessages.FAILED_DELETE_MESSAGE.format(status_code)
+                )
+            else:
```
Comment (NarenderRajuB) [inline fsd_utils/services/aws_extended_client.py:198]: remove else

File: fsd_utils/services/aws_extended_client.py [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+                        self.logger.info(
+                            f"Deleted {reciept_handles_to_delete[int(msg_meta['Id'])]}"
+                        )
+                if "Failed" in response:
+                    for msg_meta in response["Failed"]:
+                        self.logger.info(
+                            f"Could not delete {reciept_handles_to_delete[int(msg_meta['Id'])]}"
+                        )
+            return response
+        else:
```
Comment (NarenderRajuB) [inline fsd_utils/services/aws_extended_client.py:211]: Else not required here

File: fsd_utils/services/aws_extended_client.py [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+            response = self.sqs_client.delete_message_batch(
+                QueueUrl=queue_url, Entries=entries
+            )
+            # Check if the delete operation succeeded, if not raise an error?
+            status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+            if status_code != 200:
+                raise SQSExtendedClientException(
+                    ExceptionMessages.FAILED_DELETE_MESSAGE.format(status_code)
+                )
+            else:
```
Comment (NarenderRajuB) [inline fsd_utils/services/aws_extended_client.py:227]: Else not required

File: fsd_utils/services/aws_extended_client.py [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+            s3_bucket_name, s3_key = s3_details["s3BucketName"], s3_details["s3Key"]
+
+            response = self.s3_client.delete_object(Bucket=s3_bucket_name, Key=s3_key)
+            # Check if the delete operation succeeded, if not raise an error?
+            status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+            if status_code != 204:
+                raise SQSExtendedClientException(
+                    ExceptionMessages.FAILED_DELETE_MESSAGE.format(status_code)
+                )
+            else:
```
Comment (NarenderRajuB) [inline fsd_utils/services/aws_extended_client.py:263]: else not required here

**Review by NarenderRajuB — CHANGES_REQUESTED** [review]
Few comments added Please update

File: fsd_utils/services/aws_extended_client.py (hunk 1 above)
Comment (nuwan-samarasinghe) [reply fsd_utils/services/aws_extended_client.py:75]: Fixed the changes

File: fsd_utils/services/aws_extended_client.py (hunk 2 above)
Comment (nuwan-samarasinghe) [reply fsd_utils/services/aws_extended_client.py]: Updated

File: fsd_utils/services/aws_extended_client.py (hunk 3 above)
Comment (nuwan-samarasinghe) [reply fsd_utils/services/aws_extended_client.py]: Updated

File: fsd_utils/services/aws_extended_client.py (hunk 4 above)
Comment (nuwan-samarasinghe) [reply fsd_utils/services/aws_extended_client.py]: Updated

File: fsd_utils/services/aws_extended_client.py (hunk 5 above)
Comment (nuwan-samarasinghe) [reply fsd_utils/services/aws_extended_client.py]: Updated

File: fsd_utils/services/aws_extended_client.py (hunk 7 above)
Comment (nuwan-samarasinghe) [reply fsd_utils/services/aws_extended_client.py]: Updated

**Review by srh-sloan — APPROVED** [review]

**Review by NarenderRajuB — APPROVED** [review]
LGTM

**Pull request merged**